### PR TITLE
[Basic] Add fileExtension property to Path structs

### PR DESCRIPTION
**Pull request by aciidb0mb3r**
This is same as suffix but without leading `.`

### CODE REVIEW

File: Sources/Basic/Path.swift [hunk 1]
```diff
@@ -252,6 +258,12 @@ public struct RelativePath {
     public var suffix: String? {
         return _impl.suffix
     }
+
+    /// File extension of the give path's basename. This follow same rules as
+    /// suffix except that it doesn't include leading `.` character.
+    public var fileExtension: String? {
```
Comment (ddunbar) [inline Sources/Basic/Path.swift]: I think we should probably call this `extension`, or call `suffix` `fileSuffix`. @abertelrud?

**Comment by aciidb0mb3r** [comment]
@abertelrud @ddunbar what do you think about the property name? I like `fileExtension` because its clear but its a bit long. `extension` is a bit vague imo but is fine with context of paths...

**Comment by ddunbar** [comment]
Aside from name comment, LGTM but I defer to @abertelrud.

**Comment by abertelrud** [comment]
I prefer `extension`.  I do agree that either both or neither should have the word "file".

Thanks for adding this, by the way...  I considered doing it but wasn't sure since it felt like a duplication.  But I think it makes sense.

**Comment by ddunbar** [comment]
My preference is also just extension, I think.

**Comment by aciidb0mb3r** [comment]
@swift-ci please test

**Pull request merged**